--- a/dwm.c	2026-01-29 17:08:59.639611773 -0600
+++ b/dwm.c	2026-01-29 17:11:43.663063893 -0600
@@ -20,6 +20,7 @@
  *
  * To understand everything else, start reading main().
  */
+#include <X11/XF86keysym.h>
 #include <errno.h>
 #include <locale.h>
 #include <signal.h>
@@ -1806,12 +1807,17 @@
 {
 	Client *c;
 	XUnmapEvent *ev = &e->xunmap;
+	XWindowAttributes wa;
 
 	if ((c = wintoclient(ev->window))) {
 		if (ev->send_event)
 			setclientstate(c, WithdrawnState);
 		else
 			unmanage(c, 0);
+     } else if (XGetWindowAttributes(dpy, ev->window, &wa) && wa.override_redirect) {
+         /* Handle override-redirect windows (like Krita on-canvas color selector) */
+         if (selmon->sel)
+             focus(selmon->sel);
 	}
 }
 
